functions/download_recount2_data.R

```diff
@@ -0,0 +1,124 @@
+  }  
+  # Rename counts for storage
+  assign(paste0("rse_gene_", project_id), rse_gene)
+
+data_counts_rpkm <- t(rse_rpkm)
+
+## Save counts matrix to file
+write.table(data_counts_rpkm,
+          paste(local_dir,'/recount2_template_data.tsv',sep=""),
+          sep='\t',
+          row.names=TRUE,
+          col.names=NA)
+
+}
+
+
+get_recount2_compendium <- function(template_project_id,
+local_dir){
+
+  # This function downloads 50 random experiments, including the 
+  # the experiment associated with the input project_id.
+  # These experiments makeup a compendium of gene expression experiments
+  # that are saved in the local_dir
+
+  # Download metadata file
+  metadata <- all_metadata()
+  write.table(metadata, '~/Documents/Repos/generic-expression_patterns/analysis/data/metadata/recount2_metadata.tsv', sep='\t', row.names=FALSE)
+
+  # Format of the metadata
+  # Based on definitions from NCBI and blog post: 
+  # https://www.ncbi.nlm.nih.gov/books/NBK56913/#search.what_do_the_different_sra_accessi
+  # https://www.ccdatalab.org/blog/2019/3/29/gene-expression-repos-explained
+  # Project: A sequencing study (i.e. NCBI sequencing study).    
+  # Sample: Physical biospecimen on which sequencing was performed, biological source material (i.e. HeLa cell line).  
+  #         A project contains many samples
+  # Experiment: Unique sequencing library for a specific sample.  A sample can have multiple experiments, most have 1 experiment.
+  # Run: Sequencing run.  An experiment can contain many runs (i.e. technical replicates)
+  # In this case, we want to group runs into projects (for "experiment-level" simulation)
+  project_ids <- unique(metadata$project)
+
+  # Entire recount2 is 8TB
+  # We will only select the top 50 studies instead
+  selected_project_ids <- sample(project_ids, 50)
+
+  # Add user selected project id
+  selected_project_ids <- append(selected_project_ids, template_project_id)
+
+  ## Data
+
+  # Get data associated with project ids
+  # Download the RangedSummarizedExperiment object at the gene level for 
+  for(i in 1:length(selected_project_ids)){
+    if(!file.exists(file.path(selected_project_ids[i], 'rse_gene.Rdata'))) {
+      download_study(selected_project_ids[i])
+      load(file.path(selected_project_ids[i], 'rse_gene.Rdata'), verbose = TRUE)
+    } 
+    else {
+      load(file.path(selected_project_ids[i], 'rse_gene.Rdata'), verbose = TRUE)
+    }
+    # Counts are raw read counts from the sequencing run
+    # Counts are the number of reads that map that each gene
+    
+    # Scale counts by total coverage (total reads across the genome) per sample
+    # Mixed paired-end and single-end reads per run
+    # Most runs are single-end
+    # RPKM: normalizes for sequencing depth (coverge per gene) and gene length. Usually used for single-end
+    rse_rpkm <- getRPKM(rse_gene, length_var="bp_length")
+    
+    # Concatenate scaled counts into one matrix
+    if (i==1){
```

Comment:
Make sure to be consistent with the spacing of your `if`/`else` statements in these R scripts (I think the standard is a space after the `if` and before the `(`, and a space after the `)` and before the `{`).